dlrover/python/master/diagnosis/diagnosis.py

```diff
@@ -13,39 +13,61 @@
 
 import threading
 import time
+from datetime import datetime, timedelta
+from typing import Dict, List
 
+from dlrover.python.common.diagnosis import DiagnosisData
 from dlrover.python.common.log import default_logger as logger
-from dlrover.python.master.diagnosis.diagnosis_data import (
-    DataManager,
-    DiagnosisData,
-)
-from dlrover.python.master.diagnosis.diagnostician import Diagnostician
-from dlrover.python.master.diagnosis.inferencechain.common import (
+from dlrover.python.master.diagnosis.inferencechain.inference import (
     Inference,
     InferenceAttribute,
     InferenceDescription,
     InferenceName,
+    InferenceOperator,
+)
+from dlrover.python.master.diagnosis.inferencechain.inference_chain import (
+    InferenceChain,
 )
+from dlrover.python.master.diagnosis.operator.training_hang_operator import (
+    CheckTrainingHangOperator,
+)
+
+
+def has_expired(timestamp: float, time_period: int) -> bool:
+    dt = datetime.fromtimestamp(timestamp)
+    expired_dt = dt + timedelta(seconds=time_period)
+    return expired_dt < datetime.now()
 
 
 class DiagnosisManager:
     def __init__(self):
-        self.data_manager: DataManager = DataManager(600)
-        self.diagnostician: Diagnostician = Diagnostician(self.data_manager)
+        self._is_observing_started = False
+        self._data_manager: DiagnosisDataManager = DiagnosisDataManager(600)
+        self._diagnostician: Diagnostician = Diagnostician(self._data_manager)
 
     def collect_diagnosis_data(self, data_type: str, data: DiagnosisData):
-        self.data_manager.store_data(data_type, data)
+        self._data_manager.store_data(data_type, data)
+
+    def start_pre_check(self):
```

Comment:
Since pre-check is  a synchronous operation, `pre_check` or `pre_check_training` may be better.

Reply:
lgtm